### ISSUE TITLE
[iOS] Feature request: add arm64e support

### ISSUE THREAD

**Issue by Saren-Arterius** [tanersener/mobile-ffmpeg#39]
Since the introduction of Apple A12 (iPhone XS etc), arm64e is added as a new arm cpu architecture. Please include support of it if convenient.

**Comment by tanersener** [comment]
According to [Xcode 10.1 release notes](https://developer.apple.com/documentation/xcode_release_notes/xcode_10_1_release_notes?preferredLanguage=occ), `The App Store and TestFlight don't accept submissions containing arm64e. Xcode will remove arm64e content from your app when you distribute from the Organizer window. (42296212)`. 

So I think it is a bit early to publish a new release for `arm64e`, but I will add support in build scripts. Do you have an `arm64e` device, it would be great if I could test local builds on it?

**Comment by Saren-Arterius** [comment]
Hi there, currently I don't have an arm64e at the moment, but I will be
lending a iPad Pro for testing its real performance soon. You could add it
in a new branch and I will be testing it asap.

在 2018年11月8日週四 05:42，Taner Sener <notifications@github.com> 寫道：

**Comment by tanersener** [comment]
Great 👍Let me complete my local tests and I'll let you know when the branch is ready.

**Comment by tanersener** [comment]
Hi mate, I pushed my local changes under [dev-3.x-arm64e](https://github.com/tanersener/mobile-ffmpeg/tree/dev-3.x-arm64e) branch. Also published a beta release from this branch [MobileFFmpeg Full GPL v3.0 - IOS framework with arm64e beta support](https://github.com/tanersener/mobile-ffmpeg/releases/tag/v3.0.arm64e.beta). I hope you can find some time to test it on an `arm64e` device.

**Comment by Saren-Arterius** [comment]
Thanks! Currently I just got my ipad air 2 tested, but I am still
struggling to find a latest ipad pro. Let you know if I have done that.

I have a question, can I be sure that arm64e libraries will 100% be used on
arm64e supported devices? Could it fallback to arm64 for no reason? I am
afraid the test results were not done by arm64e libraries even on an ipad
pro.

在 2018年11月9日週五 03:58，Taner Sener <notifications@github.com> 寫道：

**Comment by tanersener** [comment]
Hi @Saren-Arterius. When `mobile-ffmpeg` libraries are loaded, a CPU detection module guesses the running architecture and prints it on Xcode console: `Loaded mobile-ffmpeg-x86-2.1.1`. This line shows the cpu architecture detected and loaded library version. I was expecting it to show `arm64e` but I guess detected cpu architecture does not prove that loaded binary version has the same arch.

By the way, Xcode does not install an application if it doesn't have symbols for architecture of the connected device. Did you try to install one of the `mobile-ffmpeg` releases (without `arm64e`) to that device, what Xcode does in that scenario?

**Comment by Saren-Arterius** [comment]
I don't understand. If xcode rejects to install an appication because there
is not matching architecture, how could people develop for iphone xs since
apple currently rejects the submission of that?

在 2018年11月9日週五 15:47，Taner Sener <notifications@github.com> 寫道：

**Comment by tanersener** [comment]
Let me explain:

1- Let's say you have an application A, developed in Xcode without any dependencies. And suppose that you didn't disabled any of the supported architectures under Xcode -> `Build Settings` (you build your application for all architectures). 

When you try to install this application to a device/simulator, Xcode will only build architecture of that device/simulator and run your application. Since Xcode has the sources of your application it will build appropriate binary for that device and everything will work fine. 

For example, if you use an `arm64` device, then Xcode will build binary for `arm64` architecture and run it, without any problem. If your have  Iphone XS and your Xcode version is 10.1, then `arm64e` arch will be build and installed.

2. Let's say you have an application B, developed in Xcode with `mobile-ffmpeg` pod dependency.

When you try to install this application to a device/simulator, Xcode will first build your application and then will link it to `mobile-ffmpeg` library.

For example, if you use an `arm64` device, then Xcode will first build binary for `arm64` architecture of your app and will try to link it with `mobile-ffmpeg` library. If `mobile-ffmpeg` library does not have `arm64` symbols included in its framework/universal library then your build will fail and Xcode will warn that `mobile-ffmpeg` library does not have `arm64` symbols.

This is the scenario for `arm64` or other architectures, but I'm not sure if it is the same for `arm64e`. Maybe `arm64e` has a compatibility mode to use `arm64`.

**Comment by Saren-Arterius** [comment]
Likely, nobody reported issues about could not build to iphone xs. I guess
xcode allows arm64 libraries to be used in arm64e devices.

在 2018年11月9日週五 21:43，Taner Sener <notifications@github.com> 寫道：

**Comment by tanersener** [comment]
> Likely, nobody reported issues about could not build to iphone xs. I guess xcode allows arm64 libraries to be used in arm64e devices

If you have time you can try it yourself by running test-app under `ios/test-app` folder.

**Comment by Saren-Arterius** [comment]
Yeah, just need to wait for my boss' ipad pro. Just curious about what its
real performance in ffmpeg is.

在 2018年11月9日週五 23:09，Taner Sener <notifications@github.com> 寫道：
